hack/build/repos.bzl

```diff
@@ -16,7 +16,23 @@
 
 load("@bazel_gazelle//:deps.bzl", "go_repository")
 
+# Manually defined repositories are kept separate to keep things easy to
+# maintain.
+def manual_repositories():
+    # This is needed to avoid errors when building @io_k8s_code_generator cmd/
+    # targets that look like:
+    # ERROR: /private/var/tmp/_bazel_james/7efc976e92b56f94fd5066e3c9f5d356/external/org_golang_x_tools/internal/imports/BUILD.bazel:3:1: no such package '@org_golang_x_mod//semver': The repository '@org_golang_x_mod' could not be resolved and referenced by '@org_golang_x_tools//internal/imports:go_default_library'
+    go_repository(
+        name = "org_golang_x_mod",
+        build_file_generation = "on",
+        build_file_proto_mode = "disable",
+        importpath = "golang.org/x/mod",
+        sum = "h1:KU7oHjnv3XNWfa5COkzUifxZmxp1TyI7ImMXqFxLwvQ=",
+        version = "v0.2.0",
+    )
```

Comment:
This is an odd one.. it seems like Gazelle isn't correctly picking up the module dependency on `golang.org/x/mod` automatically when grabbing it. This works around the issue for now :this_is_fine: